notes/quic.md

```diff
@@ -16,19 +16,88 @@ different streams/requests on the same connection, given one TCP connection
 you have one stream and data is sent ordered on that stream, so if you
 send multiple requests on the same TCP stream you are subject to ahead of
 line issues, where the order guarantee of the TCP stream will delay
-the receive of data from a request.
+the receive of data from a request. Since requests are orthogonal between each
+other the order guarantee of the TCP stream hurts performance.
 
-QUIC was borne as a way to be more efficient on this model where multiple
+QUIC was born as a way to be more efficient on this model where multiple
 requests are made using the same underlying connection, which is reflected
 on its design. Here I document some core design decisions that caught my
 attention, not much because they are amazing by itself, but they present
 considerable differences to TCP and may be a better fit in some scenarios
-because of it.
+because of it (the main one being "better HTTP2").
+
 
 # Built on top of UDP
 
-# Always Confidential
+This one may be accidental, since the only protocols that will work on Internet
+infrastructure are TCP and UDP if you want to have a transport protocol that
+works on the "wild" you will need to do something on top of one of these two, or
+buy a huge fight like IPv6, which so far shows how hard it is to introduce
+new protocols at Internet scale. Since networks were designed for encapsulation
+it is fairly easy to build new protocols on top of UDP/TCP. Since TCP has
+guarantees that goes against the QUIC design, you are left with UDP as an option.
+
+There are two properties that emerge from this decision that I can think so far.
+One is that you will always have the option of using your own user space
+implementation of QUIC, even when kernels already implement it on the future, which
+seems like a nice/useful option that is much harder to do with UDP/TCP.
+
+Another one is that at least for user space implementations you don't have
+limitations on how much connection you can establish per process, beyond
+the CPU/Memory, since you are going to have just one file descriptor open
+for the UDP socket and all connections/streams are built on top of that
+on user space. I have no idea how that scales, but I'm kinda curious if it would not
+be feasible in that model for a process alone manage 100K connections, something
+that is not doable in TCP given the max number of file descriptors limitation.
+
 
 # Connection identity decoupled from IP
 
+As the time I'm writing this I still don't know much details on how
+this work, I just know that QUIC connections are decoupled from IP addresses
+in the sense that the identity of a connection is not tied to IP's, instead
+of the traditional quadruple **<ipx>:<portx>:<ipy>:<porty>**.
```

Comment:
```suggestion
of the traditional quadruple `<ipx>:<portx>:<ipy>:<porty>`.
```

I think you need backticks, otherwise the < and > are interpreted as HTML.